focus/operations/src/maintenance/mod.rs

```diff
@@ -328,22 +346,106 @@ impl Runner<'_> {
     fn run_in_path(&self, time_period: TimePeriod, path: &Path) -> Result<()> {
         info!(?time_period, ?path, "running tasks");
         set_default_git_maintenance_config(path)?;
+
+        let maint_result_metric = "maint_result".to_string();
         match self.run_maint(time_period, path) {
             Ok(MaintResult::Success(status)) => {
                 if status.success() {
                     debug!(?time_period, ?path, "completed maintenance",)
                 } else {
                     warn!(?path, exit_status = ?status, "maintenance failed");
+                    let mut maint_custom_map = HashMap::new();
+                    maint_custom_map.insert(
+                        maint_result_metric,
+                        self.maint_exitstatus_metric_helper(status),
+                    );
+                    self.add_ti_invocation_message(&maint_custom_map);
                 }
             }
-            Ok(MaintResult::LockFailed) => warn!(?path, "failed to acquire lock"),
+            Ok(MaintResult::LockFailed) => {
+                warn!(?path, "failed to acquire lock");
+                let maint_custom_map =
+                    HashMap::from([(maint_result_metric, "lock_failed".to_string())]);
+                self.add_ti_invocation_message(&maint_custom_map);
+            }
             Err(e) => {
-                warn!(?path, ?e, "failed runing git-maintenance");
+                warn!(?path, ?e, "failed running git-maintenance");
+                let maint_custom_map =
+                    HashMap::from([(maint_result_metric, "git_error".to_string())]);
+                self.add_ti_invocation_message(&maint_custom_map);
```

Comment:
Be consistent about using `HashMap::from` or `let mut m = HashMap::new(); m.insert(...); m` here. If you want, you can also add the `maplit` package as a dependency (if it's not already) and write

```rust
hashmap! {
    maint_result_metric.to_string() => "git_error".to_string(),
}
```

Reply:
Fixed. Added in the maplit::hashmap